sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py

```diff
@@ -769,6 +771,10 @@ def _check_live(self, action):
     def _settle_via_mgmt_link(self, settle_operation, dead_letter_reason=None, dead_letter_description=None):
         # type: (str, Optional[str], Optional[str]) -> Callable
         # pylint: disable=protected-access
+
+        if not self._receiver:
```

Comment:
Question for @annatisch : this makes mypy happy and seems cleaner than a bunch of #ignores, and is less idiomatic than a try-catch on attribute error; but may be more targeted and clear.  

Read this as: I'm waffling on a mostly-meaningless internal change with highly subjective tradeoffs on both sides, but it's somewhat of a bellweather for how I'd tune treating similar situations elsewhere.

Reply:
(prefer keeping this based on your point of being targeted and clear)

Reply:
may I ask why we need the check here? cuz in my understanding users are not expected to create a ReceivedMessage and every ReceivedMessage are created by us linked with a receiver internally.

Reply:
As mentioned above it's _entirely_ to make mypy happy.

Basically without this check, _receiver is Optional[ServiceBusReceiver] so it doesn't like the fact that we're expecting it to be present.  I can ignore the mypy checks since bombing out on attribute error "can be fair" but there may actually be virtuous side effects of having a unified, documented, and well-established failure point, so I'm actually torn on this.